### ISSUE TITLE
error: 'certificate has expired' when using the ssl option 'Amazon RDS' 

### ISSUE THREAD

**Issue by toddcd** [mysqljs/mysql#2365]
Hi I'm trying to secure the connection between my nodejs server and an AWS RDB. I'm using the SSL profile option 'Amazon RDS' and getting a cert has expired error. The connection works perfectly SSL profile option. Is the new [AWS 2019 cert](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html) supported with the SSL profile?
```
error:
 {
    "message": "certificate has expired",
    "error": {
        "code": "HANDSHAKE_SSL_ERROR",
        "fatal": true
    }
}

my connection config config:
const aws_db = knex({
    client: 'mysql',
    connection: {
        host : AWS_DB_URL,
        user : AWS_CLIENT_USER,
        password : AWS_CLIENT_PASS,
        database : AWS_DB,
        ssl:'Amazon RDS',
    },
})
```

**Comment by dougwilson** [comment]
Yes, it is included in version 2.8.0and up. https://github.com/mysqljs/mysql/blob/master/Changes.md#v2180-2020-01-21

What version are you using?

**Comment by toddcd** [comment]
Hey Doug, thanks for the quick reply. I'm on v2.18. And if it helps, I'm running version v12.18.1 for Node and  v.021 for Knex.

here is the entire error...
Error: certificate has expired
    at TLSSocket.<anonymous> (C:\common\GitRepositories\evolytical\evolytical-common-api\node_modules\mysql\lib\Connection.js:317:48)
    at TLSSocket.emit (events.js:315:20)
    at TLSSocket._finishInit (_tls_wrap.js:936:8)
    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:710:12)
    --------------------
    at Protocol._enqueue (C:\common\GitRepositories\evolytical\evolytical-common-api\node_modules\mysql\lib\protocol\Protocol.js:144:48)
    at Protocol.handshake (C:\common\GitRepositories\evolytical\evolytical-common-api\node_modules\mysql\lib\protocol\Protocol.js:51:23)
    at Connection.connect (C:\common\GitRepositories\evolytical\evolytical-common-api\node_modules\mysql\lib\Connection.js:116:18)
    at C:\common\GitRepositories\evolytical\evolytical-common-api\node_modules\knex\lib\dialects\mysql\index.js:69:18
    at new Promise (<anonymous>)
    at Client_MySQL.acquireRawConnection (C:\common\GitRepositories\evolytical\evolytical-common-api\node_modules\knex\lib\dialects\mysql\index.js:64:12)
    at create (C:\common\GitRepositories\evolytical\evolytical-common-api\node_modules\knex\lib\client.js:291:39)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  code: 'HANDSHAKE_SSL_ERROR',
  fatal: true
}

**Comment by dougwilson** [comment]
Hi, there is no version 2.18 of this module, so it cannot be what you are using, if you are indeed using this module.

**Comment by dougwilson** [comment]
Hi, sorry, the comment I posed above was wrong, my mistake. I'm honestly not sure why you would be getting that. All of the certificates were indeed added. Unfortunately I do not have access to RDS to test. Is there a cert we are missing for you we need to add?

**Comment by toddcd** [comment]
No worries, Doug. The problem was caused by the certificate on the Amazon RDS side. I did not have access to the RDS console for this older server and assumed it had been updated. Your response got me to dig a little deeper and I found the issue, so thanks! All is good and working as expected.

https://aws.amazon.com/blogs/database/amazon-rds-customers-update-your-ssl-tls-certificates-by-february-5-2020/
